nautobot-app/{{ cookiecutter.project_slug }}/.github/workflows/release.yml

```diff
@@ -0,0 +1,103 @@
+---
+name: "Release"
+on: # yamllint disable-line rule:truthy rule:comments
+  release:
+    types: ["published"]
+
+jobs:
+  build:
+    name: "Build package with poetry"
+    runs-on: "ubuntu-latest"
+    if: "startsWith(github.ref, 'refs/tags/v')"
+    steps:
+      - uses: "actions/checkout@v4"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v6"
+        with:
+          poetry-version: "2.1.3"
+          poetry-install-options: "--dry-run"
+      - name: "Run Poetry Build"
+        run: "poetry build"
+
+      - name: "Check that the release tag matches the version in pyproject.toml"
+        run: |
+          if [ "{% raw %}${{ github.ref_name }}{% endraw %}" != "v$(poetry version -s)" ]; then exit 1; fi
+
+      - uses: "actions/upload-artifact@v4"
+        with:
+          name: "distfiles"
+          path: "dist/"
+          if-no-files-found: "error"
+
+  publish-github:
+    name: "Publish to GitHub"
+    runs-on: "ubuntu-latest"
+    if: "startsWith(github.ref, 'refs/tags/v')"
+    permissions:
+      contents: "write"
+    needs: "build"
+    steps:
+      - uses: "actions/checkout@v4"
+      - name: "Retrieve built package from cache"
+        uses: "actions/download-artifact@v4"
+        with:
+          name: "distfiles"
+          path: "dist/"
+
+      - name: "Upload binaries to release"
+        run: "gh release upload {% raw %}${{ github.ref_name }}{% endraw %} dist/*.{tar.gz,whl}"
+        env:
+          GH_TOKEN: "{% raw %}${{ secrets.GITHUB_TOKEN }}{% endraw %}"
+
+  publish-pypi:
+    name: "Push Package to PyPI"
+    runs-on: "ubuntu-latest"
+    if: "startsWith(github.ref, 'refs/tags/v')"
+    needs: "build"
+    environment: "pypi"
+    permissions:
+      # IMPORTANT: this permission is mandatory for Trusted Publishing
+      id-token: "write"
+    steps:
+      - name: "Retrieve built package from cache"
+        uses: "actions/download-artifact@v4"
+        with:
+          name: "distfiles"
+          path: "dist/"
+
+      - name: "Publish package distributions to PyPI"
+        uses: "pypa/gh-action-pypi-publish@ed0c53931b1dc9bd32cbe73a98c7f6766f8a527e"  # v1.13.0
```

Comment:
Have all of our PyPI projects been updated to support trusted publisher?
https://docs.pypi.org/trusted-publishers/adding-a-publisher/